tools/convert_datasets/a2d2.py

```diff
@@ -0,0 +1,388 @@
+        test_ratio:
+        label_suffix: Label filename ending string.
+    '''
+
+    # Create new directory structure (if not already exist)
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'img_dir'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'ann_dir'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'img_dir', 'train'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'img_dir', 'val'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'img_dir', 'test'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'ann_dir', 'train'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'ann_dir', 'val'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'ann_dir', 'test'))
+
+    # Lists containing all images and labels to symlinked
+    img_filepaths = sorted(glob.glob(osp.join(a2d2_path,
+                           '*/camera/cam_front_center/*.png')))
+    ann_filepaths = sorted(glob.glob(osp.join(a2d2_path,
+                           '*/label/cam_front_center/*{}'.format(label_suffix)
+                                              )))
+
+    # Split data according to given ratios
+    total_samples = len(img_filepaths)
+    train_ratio = 1.0 - val_ratio - test_ratio
+
+    train_idx_end = int(np.floor(train_ratio*(total_samples-1)))
+    val_idx_end = train_idx_end + int(np.ceil(val_ratio*total_samples))
+
+    # Create symlinks file-by-file
+    for sample_idx in range(total_samples):
+
+        img_filepath = img_filepaths[sample_idx]
+        ann_filepath = ann_filepaths[sample_idx]
+
+        # Partions string: [generic/path/to/file] [/] [filename]
+        img_filename = img_filepath.rpartition('/')[2]
+        ann_filename = ann_filepath.rpartition('/')[2]
+
+        if sample_idx <= train_idx_end:
+            split = 'train'
+        elif sample_idx <= val_idx_end:
+            split = 'val'
+        else:
+            split = 'test'
+
+        img_symlink_path = osp.join(a2d2_path, 'img_dir', split, img_filename)
+        ann_symlink_path = osp.join(a2d2_path, 'ann_dir', split, ann_filename)
+
+        # NOTE: Can only create new symlinks if no priors ones exists
+        try:
+            symlink(img_filepath, img_symlink_path)
+        except FileExistsError:
+            pass
+        try:
+            symlink(ann_filepath, ann_symlink_path)
+        except FileExistsError:
+            pass
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Convert A2D2 annotations to TrainIds')
+    parser.add_argument(
+        'a2d2_path', help='A2D2 segmentation data absolute path\
+                           (NOT the symbolically linked one!)')
+    parser.add_argument(
+        '-o', '--out-dir', help='Output path')
+    parser.add_argument(
+        '--convert', default=True, help='Convert label images')
+    parser.add_argument(
+        '--restruct', default=True, help='Restructure directory structure')
+    parser.add_argument(
+        '--type', default='cityscapes', help='Semantic label conversion type')
```

Comment:
We may use `choices` for this arg.

Reply:
Changed notation: Segmentation 'type' --> 'choice'